manifests/params.pp

```diff
@@ -34,24 +36,28 @@
 
   case $::osfamily {
     'RedHat': {
-      case $::operatingsystem {
-        'Fedora': {
-          if versioncmp($::operatingsystemrelease, '19') >= 0 or $::operatingsystemrelease == 'Rawhide' {
-            $provider = 'mariadb'
-          } else {
-            $provider = 'mysql'
+      if $provider_override == undef {
```

Comment:
I would be more comfortable with this if the conditional handled the provider_override case first and do some validation.

~~~
if $provider_override != undef {
  if $provider_override =~ /^(mariadb|mysql)$/ {
    $provider = $provider_override
  } else {
    # maybe error due to failed validation of $provider_override?
  }
}
~~~

I'm hesitant in keeping this vague, because for example on Fedora 19+ the default provider would be mariadb, but if the override were improperly set to `foo`, then according to this codepath, the block at line 72 would continue the setup as if it were mysql.